src/libraries/System.Private.Xml/src/System/Xml/XPath/XPathNavigator.cs

```diff
@@ -1206,7 +1206,7 @@ public virtual object Evaluate(string xpath)
             return Evaluate(XPathExpression.Compile(xpath), null);
         }
 
-        public virtual object Evaluate(string xpath, IXmlNamespaceResolver resolver)
+        public virtual object Evaluate(string xpath, IXmlNamespaceResolver? resolver)
```

Comment:
should all resolvers above also be nullable?

Reply:
Yes, and also in other places where a `IXmlNamespaceResolver` is taken, not sure if everywhere but we should be able to take null in other cases, e.g:
https://github.com/dotnet/runtime/blob/f0461cb51032e45cbcabf2152e0c82c8c5a890bc/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReader.cs#L381
The docs also mention that the param can be null https://docs.microsoft.com/dotnet/api/system.xml.xmlreader.readcontentas?view=netcore-3.1

Reply:
Do the other places `IXmlNamespaceResolver` is taken still need to be updated to be nulalble?

Reply:
`XmlReader.ReadContentAs(Type, IXmlNamespaceResolver)` is yet as not-nullable so I assume is the same for the other cases.

Reply:
I did a scan through the methods that can take `IXmlNamespaceResolver` and found a few ones that could be changed to `IXmlNamespaceResolver?`. I submitted https://github.com/dotnet/runtime/pull/41079 with the changes.